src/FormDataEncoder.ts

```diff
@@ -336,7 +337,15 @@ export class FormDataEncoder {
   async* encode(): AsyncGenerator<Uint8Array, void, undefined> {
     for (const part of this.values()) {
       if (isFile(part)) {
-        yield* part.stream()
+        const stream: any = part.stream()
+        if (stream.getReader instanceof Function) {
```

Comment:
I would rather suggest to do things the other way around here: check if `Symbol.asyncIterator` is present on `stream` object and if so, just yield the stream itself, as I originally did. Otherwise, fall back to the browser compatibility helpers.

```suggestion
        if (isFunction(stream[Symbol.asyncIterator])) {
```

Also, I would appreciate it you move this code to utility function. But I can do it myself it that bothers you.